subprojects/model-core/src/main/java/org/gradle/api/internal/provider/MappingProvider.java

```diff
@@ -28,7 +28,7 @@
     private final ProviderInternal<? extends IN> provider;
     private final Transformer<? extends OUT, ? super IN> transformer;
 
-    public MappingProvider(Class<OUT> type, ProviderInternal<? extends IN> provider, Transformer<? extends OUT, ? super IN> transformer) {
+    public MappingProvider(@Nullable Class<OUT> type, ProviderInternal<? extends IN> provider, Transformer<? extends OUT, ? super IN> transformer) {
```

Comment:
When does it happen that the type is `null`?

Reply:
This is there from my experiments with using the MappingProvider for this. This can be removed.

Reply:
Removed